server/routes/updateAlertTypeRoutes.ts

```diff
@@ -0,0 +1,51 @@
+import { Request, RequestHandler } from 'express'
+import AlertsApiClient from '../data/alertsApiClient'
+
+export default class UpdateAlertTypeRoutes {
+  constructor(private readonly alertsApiClient: AlertsApiClient) {}
+
+  public startPage: RequestHandler = async (req, res): Promise<void> => {
+    const alertTypes = (await this.alertsApiClient.retrieveAlertTypes(req.middleware.clientToken)).map(alertType => {
+      return {
+        value: alertType.code,
+        text: alertType.code,
+        hint: { text: alertType.description },
+      }
+    })
+    return res.render('pages/updateAlertType/index', { alertTypes })
+  }
+
+  public storeAlertType: RequestHandler = async (req, res): Promise<void> => {
+    const { alertType } = req.body
+    if (alertType === null || alertType === '' || alertType === undefined) {
+      const alertTypeErrorMessage = 'An alert type must be selected.'
+      const alertTypes = (await this.alertsApiClient.retrieveAlertTypes(req.middleware.clientToken)).map(at => {
+        return {
+          value: at.code,
+          text: at.code,
+          hint: { text: at.description },
+        }
+      })
+      return res.render('pages/updateAlertType/index', { alertTypes, alertTypeErrorMessage })
+    }
+    req.session.updateAlertTypeCode = alertType
+    return res.redirect('/alertType/update-description/submit-description')
+  }
+
+  public loadSubmitDescription: RequestHandler = async (req, res): Promise<void> => {
+    const alertType = await this.getAlertTypeDetails(req)
+    if (alertType === undefined) {
```

Comment:
This is an interesting check - theoretically this should never happen...the sessions are stored and backed in Redis..so if we deploy a new app then this should continue to work. @iterative-it what do you think?

Reply:
the alert type could be deactivated when user is redirected to the page?
but then i'm not filtering the alert types by `isActive`..
should that be a criteria?

Reply:
I think it makes sense that we only display active alerts to be updated. I think we should then reactivate deactivated alert codes and types - and then update them....I'd much rather we stick to one action per journey.

Reply:
As an aside, I think you can use `if(!alertType)` in place of the various checks for null and undefined but that's not really important.

This is one of those theoretical possibilities that you'd cover for code completeness that would never actually happen in real life. In this case because the functionality will only have one user - a nominated system admin.

If the code was in the URL, I'd be inclined to keep this to protect the user from shared or badly entered URLs. As it's in session and will have been correct when selected, I'd probably remove it. The API should also enforce any checks on the existence of the alert code and whether it's in a state to be updated and that can be shown when the user tries to action the update.

If we want to check that the state of an alert code is correct, we'd want to do that everywhere we're in the context of an alert code. In that case middleware would be a better choice.